src/context.rs

```diff
@@ -16,6 +16,20 @@ type Callback = dyn FnOnce(&mut Context);
 /// A handler for an event type `E`
 type EventHandler<E> = dyn Fn(&mut Context, E);
 
+/// An enum to indicate the priority for plans at a given time.
+///
+/// Most plans will have `Normal` priority. Plans with priority `First` are
+/// handled before all `Normal` plans, and those with priority `Last` are
+/// handled after all `Normal` plans. In all cases ties between plans at the
+/// same time and with the same priority are handled in the order of scheduling.
+///
+#[derive(PartialEq, Eq, Ord, PartialOrd)]
+pub enum PlanPriority {
```

Comment:
I'm sorry to nitpick about words but I really don't like the term "Priority" here which implies like some unbounded arbitrary numbering system (which I don't think we want to generally encourage).  Can we do maybe `PlanPhase` or `ExecutionPhase` and refer to them in docs/comments as "phases"

Reply:
Yes! Totally fair. Switched to `ExecutionPhase` -- thanks!